src/Controls/src/Core/NavigationModel.cs

```diff
@@ -162,8 +177,21 @@ public void Push(Page page, Page ancestralNav)
 
 		public void PushModal(Page page)
 		{
+			var previousPage = CurrentPage;
 			_navTree.Add(new List<Page> { page });
 			_modalStack.Add(page);
+
+			// Shell handles its own page life cycle events
+			// because you can push multiple pages in a single
+			// request
+			if (_navTree.Count > 0 &&
+				_navTree[0].Count > 0 &&
+				_navTree[0][0] is not Shell)
+			{
+				previousPage.SendNavigatingFrom(new NavigatingFromEventArgs());
```

Comment:
How about having a static `NavigatingFromEventArgs.Empty`? Just like `EventArgs.Empty` this will prevent us from instantiate a new `NavigatingFromEventArgs` every time that we pass here and at line `107`